### DISCUSSION TITLE
How to open "dialog with message" from Rust?

### DISCUSSION THREAD

**Discussion by EGSP** [tauri-apps/tauri#9642] [Q&A] [answered]
I want to show error message from Rust to explain problem to user. Now im using panic, but it just crashes app, as expected i think.

I dont understand Tauri API docs. Moreover, it seems that Tauri API shows JavaScript frontend code. 

Can someone show example of code which function and how to call from Rust?

**Comment by FabianLars** [comment, marked as answer]
The rust documentation is on docs.rs: https://docs.rs/tauri/latest/tauri/api/dialog/fn.message.html / https://docs.rs/tauri/latest/tauri/api/dialog/blocking/fn.message.html

The docs are generally stil lacking but in this case i think there isn't that much to show. Here's an in-code example (for ask but message is almost the same) https://github.com/tauri-apps/tauri/blob/b02fc90f450ff9e9d8a35ee55dc1beced4957869/examples/api/src-tauri/src/main.rs#L224-L236

**Reply by EGSP** [reply]
When to use blocking dialog?

**Reply by FabianLars** [reply]
Basically only in async tauri commands (or async functions in general). non-async commands, and the Builder's hooks (like .setup) will run on the main thread which will require the non-blocking dialog.

**Reply by EGSP** [reply]
Thank you for clarification!

What should i do to enable dialogs api in my project?
![image](https://github.com/tauri-apps/tauri/assets/34096047/06e8dc0d-c442-4df4-ad9c-b9310c7e2e0d)
![image](https://github.com/tauri-apps/tauri/assets/34096047/1dad67db-0db0-4c03-971f-ffb948e9c33f)
![image](https://github.com/tauri-apps/tauri/assets/34096047/5c73c94d-0747-4e50-9115-fce9e53f9e80)

**Reply by EGSP** [reply]
Nevermind, found answer in JavaScript Tauri api
![image](https://github.com/tauri-apps/tauri/assets/34096047/cdebf37f-089d-4cce-ace0-bf46962da51a)
